sourcerer/parser.cs

```diff
@@ -214,16 +216,67 @@ public class Function : Declaration {
   }
 
   public class Include : Node {
-    public Include(string text, Node parent, string[] path) :
-        base(text, parent) {
+    // The |file_info| is the file where this include appears, not the one that
+    // is included.
+    public Include(Node? parent, string[] path, FileInfo file_info) :
+        this(text: null, parent, path, file_info) { }
+
+    public Include(string? text,
+                   Node? parent,
+                   string[] path,
+                   FileInfo file_info)
+        : base(text, parent) {
       this.path = path;
+      file_info_ = file_info;
+      own_body_ = Regex.Replace(file_info_.Name,
+                                @"(?<!_body)\.hpp$",
+                                "_body.hpp");
+      own_header_ = Regex.Replace(file_info_.Name,
+                                  @"(_body|_test)?\.[hc]pp$",
+                                  ".hpp");
+    }
+
+    public override string Cxx() {
+      Debug.Assert(must_rewrite, "inconsistent rewrite");
+      return "#include \"" + string.Join('/', path) + ".hpp\"";
     }
 
     public override void WriteNode(string indent = "") {
       Console.WriteLine(indent + "Include " + string.Join(", ", path));
     }
 
     public string[] path;
+
+    public bool is_blessed_by_sourcerer =>
+        text.EndsWith("// 🧙");
```

Comment:
We should be able to add an explanation. Technically we can do that by having
`// Needed for XYZ // 🧙`,
but maybe EndsWith("🧙") would be enough (alternatively, a regex so it can be at the beginning of the comment.

Reply:
Done.